### ISSUE TITLE
GitHub language support for Just/Justfiles

### ISSUE THREAD

**Issue by mihaigalos** [casey/just#1347]
Hi,

I love `just` but am missing syntax coloring/language support for it on the GitHub side.
PR https://github.com/github/linguist/pull/6086 is a draft and aims at enabling that.

@skellock, @casey, can you please conduct an initial review of the it? I'm adding @skellock first since the source rendering for Justfiles is present at https://github.com/skellock/vscode-just.

Would appreciate any help in merging the PR or otherwise just enabling Justfile support in GitHub.

**Comment by casey** [comment]
Awesome, thanks so much for taking the initiative for doing this! I had a look at the PR, and it looks good to me.

**Comment by mihaigalos** [comment]
Hi @casey. GH Linguist release [`v7.24.0`](https://github.com/github/linguist/releases/tag/v7.24.0) has been rolled out.
I already saw the first [`Justfile`](https://github.com/mihaigalos/config/blob/main/Justfile) correctly rendered (see below).
I believe we can close this issue now, don't you?

![image](https://user-images.githubusercontent.com/16443090/202023508-70b618fd-7a79-4726-97df-523ee28a89aa.png)

**Comment by casey** [comment]
Nice! I should have thought of this before, but does it work with any case-insensitive match of the text `justfile`, or only `Justfile`?

**Comment by casey** [comment]
Whoops, accidentally closed.

**Comment by mihaigalos** [comment]
> I should have thought of this before, but does it work with any case-insensitive match of the text `justfile`, or only `Justfile`?

Yes [you should have](https://github.com/github/linguist/pull/6086#discussion_r975085559) 😅.
Please open a PR adding other potential names of files [here](https://github.com/github/linguist/pull/6086/files#diff-9c337ec09ccaaa1374713aad9ba26cdc1308586a27cb309a883a0b19bde663d8R7693).

**Comment by casey** [comment]
Sounds good! Just opened [this PR](https://github.com/github/linguist/pull/6167).

**Comment by nk9** [comment]
Over the past couple of months, I've done a bunch of work to improve the Just syntax. The work has been done on the Sublime syntax definition that's in Just's `extras` folder. At this point, I've completely re-written it, and I'll shortly be releasing the new syntax as a Sublime package.

I believe Linguist accepts `.sublime-syntax` files. It's fortuitous that this push to include Just syntax in GitHub is happening now; I didn't even know about this work when I started hacking on the syntax! But once the new syntax is ready to go, what steps would be necessary to begin using that in Linguist?

Happy to have that discussion in a different bug as needed, I just don't know where to go at this point.

**Comment by mihaigalos** [comment]
@nk9, have a look at the referenced PR in this issue. The PR is in [Linguist](https://github.com/github/linguist), as you pointed out.
That repo contains all the docs you need to add future languages.

**Comment by nk9** [comment]
Thanks for the quick reply! My question is more about replacing the VSCode syntax used in the Linguist language you've already created. Creating a new language would cause a conflict, I would think.

Maybe the answer is just "submit a PR to Linguist to replace the syntax once it's ready," but since you're leading on the Linguist integration I wanted to check with you first. Thanks!

**Comment by mihaigalos** [comment]
The current Linguist support for Justfiles is based on [skellock/vscode-just](https://github.com/skellock/vscode-just), which is archived.

Also, some features don't fully work, see [this](https://github.com/mihaigalos/config/blob/8ba7b926113c215ee695acc2aac71e372437a7f0/services/ntfy/Justfile#L11), with the not properly rendered result:
![image](https://user-images.githubusercontent.com/16443090/208047120-694e49e7-914c-4441-8c91-a5adf6b24e78.png).

It would be massive if you could contribute an improved approach. Let us know how to help.

**Comment by casey** [comment]
I think this issue can be closed, since GitHub now ships a version of linguist that supports Just. Feel free to keep discussing here or in another issue on updating the syntax file!
